presentation/common/src/test/content/Descriptor.test.ts

```diff
@@ -87,21 +88,77 @@ describe("Descriptor", () => {
       expect(descriptor).to.matchSnapshot();
     });
 
-    it("creates valid descriptorJSON from CompressedDescriptorJSON with nestedField", () => {
-      const compressedDescriptorJSON = compressDescriptorJSON(testDescriptorJSON);
-      const descriptorFromCompressedJSON = Descriptor.fromJSON(JSON.stringify(compressedDescriptorJSON));
-      const descriptor = Descriptor.fromJSON(JSON.stringify(testDescriptorJSON));
-      expect(descriptor).to.deep.eq(descriptorFromCompressedJSON);
+    it("creates valid Descriptor from CompressedDescriptorJSON", () => {
+      const categories = [createRandomCategoryJSON(), createRandomCategoryJSON(), createRandomCategoryJSON(), createRandomCategoryJSON()];

```

Comment:
Why not just use one category for everything? We're not interested in categories' handling, so there should be as little code as possible related to them.